apis/kueue/v1beta1/workload_types.go

```diff
@@ -302,6 +302,11 @@ const (
 	// more detailed information. The more detailed reasons should be prefixed
 	// by one of the "base" reasons.
 	WorkloadPreempted = "Preempted"
+
+	// WorkloadRequeued means that the Workload was requeued.
```

Comment:
Please make sure we have an integration test which checks that:
- `Requeued=True` when `QuotaReserved=True` and `Admitted=False`
- `Requeued=False` when the workload is evicted (IIUC this may still need implementation)

Reply:
> Requeued=False when the workload is evicted (IIUC this may still need implementation)

Could you please explain this case? Is it possible to set Requeued to false?

Reply:
OTOH, setting it to `False` is not required to compute the metric, so maybe it does not need to be in this PR. WDYT @alculquicondor ?

Reply:
As discussed offline, we don't want to change the condition to False on admission.

Reply:
The condition should transition to `False` in the following scenarios:
- When workload.spec.active=false. It should return to `True` when it's reactivated.
- When evicted due to WaitForPodsReady, the workload will temporarily be in backoff. During this time, `Requeued` should be false.

Although I think we can leave these for a follow up.